atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/health/MetricsBasedTimelockHealthCheckTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.health;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import com.palantir.atlasdb.transaction.impl.InstrumentedTimelockService;
+import com.palantir.atlasdb.util.AtlasDbMetrics;
+import com.palantir.lock.v2.TimelockService;
+
+public class MetricsBasedTimelockHealthCheckTest {
```

Comment:
Please add test cases for "one success after multiple failures => unhealthy", and "multiple successes and one failure => healthy"

It might be easier to do these by mocking two `TimelockService` endpoints, and having one succeed and the other fail.

Reply:
Done!